packages/unified-selection/learning/SelectionScopes.md

```diff
@@ -0,0 +1,37 @@
+- `functional` - return key of element's related functional element. For `BisCore.GeometricElement3d` the related functional element is found using the `Functional.PhysicalElementFulfillsFunction` relationship. For `BisCore.GeometricElement2d` the nearest functional element is searched for using the `Functional.DrawingGraphicRepresentsFunctionalElement` relationship - if the given element has a related functional element, it will be returned, otherwise the element's parent will be checked and if it also does not have a related functional, then the parent of the parent will be checked until no more ancestors can be traversed or a functional element is found. Regardless whether it is an `BisCore.GeometricElement2d` or `BisCore.GeometricElement3d` if no functional element is found then the element itself will be returned.
+
+The `element` and `functional` scopes also support specifying an `ancestorLevel` property, which specifies how far "up" we should walk to find the target element. When not specified or `0`, the target element matches the request element. When set to `1`, the target element matches the direct parent element. When set to `2`, the target element is parent of the parent element and so on. In all situations when this is `> 0`, we're not walking further than the last existing element, for example when `ancestorLevel = 1` (direct parent element is requested), but the request element doesn't have a parent, the request element is returned as the result. A negative value would result in the top-most element to be returned.
+
+The `@itwin/unified-selection` package delivers a `computeSelection` function for computing which elements should be added into unified selection storage based on the given element ID's and a specified selection scope:
+
+```ts
+import { computeSelection } from "@itwin/unified-selection";
+import { IModelConnection } from "@itwin/core-frontend";
+import { createECSqlQueryExecutor } from "@itwin/presentation-core-interop";
+const queryExecutor = createECSqlQueryExecutor(imodel);
+const selection = computeSelection({ queryExecutor, elementIds, scope: "element" });
+```
+
+`element` and `functional` scopes additionally allow selecting assembly elements by specifying the `ancestorLevel` property in the selection scope argument of `computeSelection` function. The `ancestorLevel` property specifies how far "up" we should walk to find the target element. When not specified or `0`, the target element matches the request element. When set to `1`, the target element matches the direct parent element. When `2`, the target element is the parent of the parent element, and so on. In all situations when this is `> 0`, we're not walking further than the last existing element, for example, when `ancestorLevel = 1` (direct parent element is requested), but the request element doesn't have a parent, the request element is returned as the result. A negative value would result in the top-most element to be returned.
```

Comment:
This paragraph is the same as the previous one. Is this intentional?

Reply:
nm, this is a result of me moving things around... removed one of these paragraphs